components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,666 @@
+    :ref:`method-resolver`.
+
+Setting Rule
+~~~~~~~~~~~~
+
+Every shipping method can have many rules which defines additional requirements for shipment. If shipment do not
+fulfil this requirements (for example rule sets expected quantity of shipment items to 2, but shipment has only one **ShipmentItem**)
+then shippingMethod with this rule cannot be used to this shipment.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Rule;
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+
+        $shippingMethod = new ShippingMethod();
+        $rule = new Rule();
+
+        $shippingMethod->addRule($rule);
+        $shippingMethod->hasRule($rule); // returns true
+        $shippingMethod->getRules(); // collection of rules with count equals 1
+        $shippingMethod->removeRule($rule);
+        $shippingMethod->hasRule($rule); // returns false
+
+Shipping Method Translation
+---------------------------
+
+**ShippingMethodTranslation** allows translate name of shipping method according to the given locales. To see how to use translation
+please go to :ref:`component_translation_basic-translations`.
+
+Rule
+----
+
+**Rule** defines additional requirements for shipment method which should be fulfil by shipment if it have to be delivered in
+the way which is described in shipping method.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Rule;
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Model\RuleInterface;
+
+        $shippingMethod = new ShippingMethod();
+        $rule = new Rule();
+        $rule->setConfiguration(array('count' => 1, 'equal' => true));
+        $rule->setType(RuleInterface::TYPE_ITEM_COUNT);
+        $shippingMethod->addRule($rule);
+
+
+Shipment Item
+-------------
+
+You can use shipment item to connecting shippable object with proper shipment.
+Shipping item can exist without assigned shipment.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Shipment;
+        use Sylius\Component\Shipping\Model\ShipmentItem;
+        use Sylius\Component\Shipping\Model\ShipmentInterface;
+
+        $shipment = new Shipment();
+        $shippable = new ShippableObject(); // class which implements ShippableInterface
+        $shipmentItem = new ShipmentItem();
+
+        $shipmentItem->setShipment($shipment);
+        $shipmentItem->getShipment(); // returns shipment object
+        $shipmentItem->setShipment();
+        $shipmentItem->getShipment(); // returns null
+
+        $shipmentItem->setShippable($shippable);
+        $shipmentItem->getShippable(); // returns shippable object
+
+        $shipmentItem->getShippingState(); // returns const STATE_READY
+        $shipmentItem->setShippingState(ShipmentInterface::STATE_BACKORDERED);
+        $shipmentItem->getShippingState(); // returns const STATE_BACKORDERED
```

Comment:
Again, no need to be that detailed. Show get/set and should be pretty obvious what happens.